pydocx/tests/test_html.py

```diff
@@ -4,15 +4,268 @@
     unicode_literals,
 )
 
-from pydocx.tests import DocumentGeneratorTestCase
-from pydocx.wordml import ImagePart
+from pydocx.tests import (
+    DocumentGeneratorTestCase,
+    WordprocessingDocumentFactory,
+)
+from pydocx.wordml import (
+    FootnotesPart,
+    ImagePart,
+    MainDocumentPart,
+    StyleDefinitionsPart,
+)
+
+
+class FootnoteTestCase(DocumentGeneratorTestCase):
+    def test_footnote_without_definition_is_ignored(self):
+        document_xml = '''
+            <p>
+              <r>
+                <t>Foo</t>
+              </r>
+              <r>
+                <footnoteReference id="abc"/>
+              </r>
+            </p>
+        '''
+        document = WordprocessingDocumentFactory()
+        document.add(MainDocumentPart, document_xml)
+
+        expected_html = '<p>Foo</p>'
+        self.assert_document_generates_html(document, expected_html)
+
+    def test_basic_footnote_with_styling(self):
+        document_xml = '''
+            <p>
+              <r>
+                <t>Foo</t>
+              </r>
+              <r>
+                <rPr>
+                  <vertAlign val="superscript"/>
+                </rPr>
+                <footnoteReference id="abc"/>
+              </r>
+            </p>
+            <p>
+              <r>
+                <t>Footnotes should appear below this</t>
+              </r>
+            </p>
+        '''
+
+        footnotes_xml = '''
+            <footnote id="abc">
+              <p>
+                <r>
+                  <rPr>
+                    <b val="on"/>
+                  </rPr>
+                  <footnoteRef/>
+                  <t>Bar</t>
+                </r>
+              </p>
+            </footnote>
+        '''
+
+        document = WordprocessingDocumentFactory()
+        document.add(FootnotesPart, footnotes_xml)
+        document.add(MainDocumentPart, document_xml)
+
+        expected_html = '''
+            <p>
+                Foo
+                <sup>
+                    <a href="#footnote-abc" name="footnote-ref-abc">1</a>
+                </sup>
+            </p>
+            <p>Footnotes should appear below this</p>
+            <hr/>
+            <ol type="1">
```

Comment:
I did this because I thought it's what you had suggested. I can change it to whatever makes the most sense.